client/filters.go

```diff
@@ -0,0 +1,46 @@
+package client
+
+import (
+	"encoding/json"
+	"net/url"
+)
+
+// Filters describes a predicate for an API request.
+//
+// Each entry in the map is a filter term.
+// Each term is evaluated against the set of values.
+// A filter term is satisfied if any one of the values in the set is a match.
+// An item matches the filters when all terms are satisfied.
+//
+// Like all other map types in Go, the zero value is empty and read-only.
+type Filters map[string]map[string]bool
+
+// Add appends values to the value-set of term.
+//
+// The receiver f is returned for chaining.
+//
+//	f := make(Filters).Add("name", "foo", "bar").Add("status", "exited")
+func (f Filters) Add(term string, values ...string) Filters {
+	if _, ok := f[term]; !ok {
+		f[term] = make(map[string]bool)
+	}
+	for _, v := range values {
+		f[term][v] = true
+	}
+	return f
+}
```

Comment:
`Add` not only _adds_, but _replaces_ what was already there; or at least for the `term` key, it's, it would keep it as-is, but for `values`, it replaces what was there.


Wondering if `Set` would be a more appropriate name? Alternatively `AddValues` I guess could be an option.

Reply:
No, `Add` appends, same as [`(url.Values).Add`](https://pkg.go.dev/net/url#Values.Add). I'll cover that in the unit test.